base_location/models/res_city.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2018 Aitor Bouzas <aitor.bouzas@adaptivecity.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class City(models.Model):
+    _inherit = 'res.city'
+
+    zip_ids = fields.One2many('res.city.zip', 'city_id',
+                              string="Zips in this city")
+
+    _sql_constraints = [
+        ('name_state_country_uniq',
+         'UNIQUE(name, state_id, country_id)',
+         'You already have a city with that name in the same state.'
```

Comment:
make the error translatable

Reply:
I would say that it's translatable this way already... Odoo takes care of this. If not, you will have static translations at the server startup.